pkg/resourcecreator/istio_test.go

```diff
@@ -130,20 +130,39 @@ func TestIstio(t *testing.T) {
 		assert.NotContains(t, serviceRoleBinding.Spec.Subjects, &subject)
 	})
 
-	t.Run("prometheus enabled adds istio system default service account to servicerolebinding", func(t *testing.T) {
+	t.Run("no service role and no service role binding created for prometheus, when disabled ", func(t *testing.T) {
+		app := fixtures.MinimalApplication()
+		app.Spec.Prometheus.Enabled = false
+		err := nais.ApplyDefaults(app)
+		assert.NoError(t, err)
+
+		serviceRolePrometheus, err := resourcecreator.ServiceRolePrometheus(app)
+		assert.NoError(t, err)
+		assert.Nil(t,serviceRolePrometheus)
+
+		serviceRoleBindingPrometheus, err := resourcecreator.ServiceRoleBindingPrometheus(app)
+		assert.NoError(t, err)
+		assert.Nil(t, serviceRoleBindingPrometheus)
+
+	})
+
+	t.Run("service role and service role binding created, with matching naming. When prometheus is enabled", func(t *testing.T) {
 		app := fixtures.MinimalApplication()
 		app.Spec.Prometheus.Enabled = true
+		app.Spec.AccessPolicy.Ingress.AllowAll = true
 		err := nais.ApplyDefaults(app)
 		assert.NoError(t, err)
 
-		serviceRoleBinding, err := resourcecreator.ServiceRoleBinding(app)
+
+		serviceRolePrometheus, err := resourcecreator.ServiceRolePrometheus(app)
 		assert.NoError(t, err)
-		assert.NotNil(t, serviceRoleBinding)
+		assert.NotNil(t,serviceRolePrometheus)
```

Comment:
this file needs formatting

Reply:
Blame the IDE. Hadn't configured formatter for Goland my bad